crates/loans/src/lib.rs

```diff
@@ -1781,20 +1710,24 @@ impl<T: Config> Pallet<T> {
     }
 
     /// Get the total balance of `who`.
+    /// Ignores any frozen balance of this account.
     fn balance(asset_id: AssetIdOf<T>, who: &T::AccountId) -> BalanceOf<T> {
         <orml_tokens::Pallet<T> as MultiCurrency<T::AccountId>>::total_balance(asset_id, who)
     }
 
-    /// Total supply of lending tokens (ptokens), given the underlying
+    /// Total supply of lending tokens (lend_tokens), given the underlying
     pub fn total_supply(asset_id: AssetIdOf<T>) -> Result<BalanceOf<T>, DispatchError> {
-        let ptoken_id = Self::ptoken_id(asset_id)?;
-        Ok(orml_tokens::Pallet::<T>::total_issuance(ptoken_id))
+        let lend_token_id = Self::lend_token_id(asset_id)?;
+        Ok(orml_tokens::Pallet::<T>::total_issuance(lend_token_id))
     }
 
-    /// Free lending tokens (ptokens) of an account, given the underlying
-    pub fn free_ptokens(asset_id: AssetIdOf<T>, account_id: &T::AccountId) -> Result<Amount<T>, DispatchError> {
-        let ptoken_id = Self::ptoken_id(asset_id)?;
-        let amount = Amount::new(orml_tokens::Pallet::<T>::free_balance(ptoken_id, account_id), ptoken_id);
+    /// Free lending tokens (lend_tokens) of an account, given the underlying
+    pub fn free_lend_tokens(asset_id: AssetIdOf<T>, account_id: &T::AccountId) -> Result<Amount<T>, DispatchError> {
+        let lend_token_id = Self::lend_token_id(asset_id)?;
```

Comment:
Is it worth returning an error if `asset_id` is already a `LendToken`?

Reply:
An error is returned if the `asset_id` is already a LendToken: `Error::<T>::MarketDoesNotExist` (as in, there is no market for lending out the given asset, so a LendToken for the `asset_id` doesn't exist)